src/ngraph_builder.cc

```diff
@@ -41,6 +41,36 @@ const static std::map<tf::DataType, ngraph::element::Type> TF_NGRAPH_TYPE_MAP =
      {tf::DataType::DT_UINT16, ng::element::u16},
      {tf::DataType::DT_BOOL, ng::element::boolean}};
 
+// Helper for storing ops in ng_op_map
+static void SaveNgOp(Builder::OpMap& map, std::string key, 
+                     shared_ptr<ng::Node> value, 
+                     size_t index = 0) { 
+  if (map[key].size() <= index) {
+    map[key].resize(index + 1);
+  }
+  map[key][index] = value; 
+}
+
+// Helper for lookup in ng_op_map 
+static shared_ptr<ng::Node> GetNgOp(Builder::OpMap &map, 
+                                    std::string key, 
+                                    size_t index = 0) {
+  return map.at(key).at(index);
```

Comment:
So - when the lookup fils - this will throw an exception - please catch it. 
Also - pass the map as a `const` reference as you are not modifying this. 
Finally - since the value is vector - you could use `[]` as opposed to `at` - no?

Reply:
1) So we will return here `tensorflow::Status` and add parameter for extracted value, then using it like TF_RETURN_IF_ERROR(GetNgOp(result, key, index)) - sounds like a plan 
2) Good point
3) `[]` does not have bounds checking, so lookup may just crash app.  `at()` will throw `out_of_bounds`

Reply:
Good point re. #3